src/averageAges.js

```diff
@@ -37,7 +42,24 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  if (!withChildren) {
+    return getAverageAge(women);
+  }
+
+  const mothers = women.filter(woman => people
+    .some(person => person.mother === woman.name)
+  );
+
+  return getAverageAge(mothers);
+
+  function getAverageAge(womenArr) {
+    const totalAge = womenArr
+      .reduce((acc, person) => acc + person.died - person.born, 0);
+
+    return totalAge / womenArr.length;
```

Comment:
move this function in the upper scope and use it in calculateMenAverageAge function too